lib/spack/docs/build_systems/rpackage.rst

```diff
@@ -0,0 +1,338 @@
+
+   ``Depends`` is required and will cause those R packages to be *attached*,
+   that is, their APIs are exposed to the user. ``Imports`` *loads* packages
+   so that *the package* importing these packages can access their APIs,
+   while *not* being exposed to the user. When a user calls ``library(foo)``
+   s/he *attaches* package ``foo`` and all of the packages under ``Depends``.
+   Any function in one of these package can be called directly as ``bar()``.
+   If there are conflicts, user can also specify ``pkgA::bar()`` and
+   ``pkgB::bar()`` to distinguish between them. Historically, there was only
+   ``Depends`` and ``Suggests``, hence the confusing names. Today, maybe
+   ``Depends`` would have been named ``Attaches``.
+
+   The ``LinkingTo`` is not perfect and there was recently an extensive
+   discussion about API/ABI among other things on the R-devel mailing
+   list among very skilled R developers:
+
+   * https://stat.ethz.ch/pipermail/r-devel/2016-December/073505.html
+   * https://stat.ethz.ch/pipermail/r-devel/2017-January/073647.html
+
+Some packages also have a fourth section:
+
+* Suggests
+
+These are optional, rarely-used dependencies that a user might find
+useful. You should **NOT** add these dependencies to your package.
+R packages already have enough dependencies as it is, and adding
+optional dependencies can really slow down the concretization
+process. They can also introduce circular dependencies.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Core, recommended, and non-core packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you look at "Depends", "Imports", and "LinkingTo", you will notice
+3 different types of packages:
+
+"""""""""""""
+Core packages
+"""""""""""""
+
+If you look at the ``caret`` homepage, you'll notice a few dependencies
+that don't have a link to the package, like ``methods``, ``stats``, and
+``utils``. These packages are part of the core R distribution and are
+tied to the R version installed. You can basically consider these to be
+"R itself". These are so essential to R so it would not make sense that
+they could be updated via CRAN. If so, you would basically get a different
+version of R. Thus, they're updated when R is updated.
+
+You can find a list of these core libraries at:
+https://github.com/wch/r-source/tree/trunk/src/library
+
+""""""""""""""""""""
+Recommended packages
+""""""""""""""""""""
+
+When you install R, there is an option called ``--with-recommended-packages``.
+This flag causes the R installation to include a few "Recommended" packages
+(legacy term). They are for historical reasons quite tied to the core R
+distribution, developed by the R core team or people closely related to it.
+The R core distribution "knows" about these package, but they are indeed
+distributed via CRAN. Because they're distributed via CRAN, they can also be
+updated between R version releases.
+
+Spack explicitly adds the ``--without-recommended-packages`` flag to prevent
```

Comment:
This is not actually true at the moment, the PR that introduced this was abandoned due to merge conflicts. We really need to make this happen if we want reliable R activation support.